src/components/Microsoft.DotNet.UpgradeAssistant/Checks/VisualBasicRazorTemplateCheck.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Checks
+{
+    /// <summary>
+    /// Users running upgrade-assistant on ASP.NET projects with VBHTML
+    /// templates should be informed that this is an unsupported technology
+    /// before they invest significant time running the tool.
+    /// </summary>
+    public class VisualBasicRazorTemplateCheck : BypassableReadinessCheck
+    {
+        public VisualBasicRazorTemplateCheck(ILogger<VisualBasicRazorTemplateCheck> logger, UpgradeOptions upgradeOptions)
+            : base(logger, upgradeOptions)
+        {
+        }
+
+        /// <summary>
+        ///  Gets the value VisualBasicRazorTemplateCheck.
+        /// </summary>
+        public override string Id => nameof(VisualBasicRazorTemplateCheck);
+
+        /// <summary>
+        /// Gets a value indicating whether this check can be bypassed using the ReadinessAcknowledgement option.
+        /// </summary>
+        public override bool IsBypassable => true;
+
+        public override string TechnologyDetected => "VB Razor Engine";
+
+        // TODO: code review - do we expect people to copy and paste this? Maybe we need to make a friendly URL.
```

Comment:
Friendly URL would probably be better (it would also enable us to redirect to other docs if better ones become available without changing UA). That said, I think it's not a high priority and this is probably fine for the time being to get v1 of the feature merged.